transport/session_test.go

```
@@ -501,3 +501,22 @@ func TestHandlePackageWithNilListenerDoesNotPanicOnError(t *testing.T) {
 
 	ss.handlePackage()
 }
+
+// Regression test for #122: promoted Connection methods must not panic after
+// the session has cleared its embedded Connection via gc().
+func TestSetMethodsAfterGCDoNotPanic(t *testing.T) {
+	c1, c2 := net.Pipe()
+	defer c1.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `c1.Close` is not checked

(errcheck)